File: qryoklama_flutter/lib/service/api_client.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class ApiClient {
  Future<http.Response> postJson(String url, Map<String, dynamic> body,
      {Map<String, String>? headers}) {
    return http.post(
      Uri.parse(url),
      headers: {
        "Content-Type": "application/json; charset=utf-8",
        "X-Platform": "flutter",
        ...?headers,
      },
      body: jsonEncode(body),
    );
  }

  Future<http.Response> get(String url, {Map<String, String>? headers}) {
    return http.get(Uri.parse(url), headers: headers);
  }
}

File: qr_yoklama_react_native/src/service/apiClient.ts
export class ApiClient {
    async get(url: string): Promise<Response> {
        return fetch(url, { method: "GET" });
    }

    async postJson(url: string, body: Record<string, unknown>): Promise<Response> {
        return fetch(url, {
            method: "POST",
            headers: {
                "Content-Type": "application/json; charset=utf-8",
                "X-Platform": "expo",
            },
            body: JSON.stringify(body),
        });
    }
}
